KoreBuild-dotnet/build/_k-standard-goals.shade

```diff
@@ -179,12 +179,49 @@ default NUGET_FEED = 'https://api.nuget.org/v3/index.json'
 
 #xunit-test target='test' if='Directory.Exists("test")'
   @{
-    var projectFiles = Files.Include("test/**/project.json").Exclude("**/bin/*/app/project.json").ToList();
-    projectFiles.ForEach(projectFile => XunitTest(projectFile, testParallel: false));
-    projectFiles.ForEach(projectFile => DotnetTest(projectFile, testParallel: false));
-    projectFiles.ForEach(projectFile => DnxTest(projectFile, testParallel: false));
+    var projectFiles = Files.Include("test/**/project.json").Exclude("**/bin/*/app/project.json");
+    foreach (var projectFile in projectFiles)
+    {
+        var projectText = File.ReadAllText(projectFile);
+        var project = (JsonObject)Json.Deserialize(projectText);
+        var configs = project.ValueAsJsonObject("frameworks");
+        var targetFrameworks = configs == null ? new string[0] : configs.Keys;
+
+        var net4xTFM = targetFrameworks.FirstOrDefault(t => t.StartsWith("net4", StringComparison.OrdinalIgnoreCase));
+        var dnx451TFM = targetFrameworks.FirstOrDefault(t => t.Equals("dnx451", StringComparison.OrdinalIgnoreCase));
+        var dnxCore50TFM = targetFrameworks.FirstOrDefault(t => t.Equals("dnxcore50", StringComparison.OrdinalIgnoreCase));
+
+        if (dnxCore50TFM != null)
+        {
+            if (projectText.Contains("dotnet-test-xunit"))
+            {
+                DotnetTest(projectFile, dnxCore50TFM,  testParallel: false);
```

Comment:
Ehh, it's figuring out two conditions (some of which will eventually get removed). Don't see a great deal of value in flowing all of these through.